test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -145,20 +145,39 @@ public void execute_add_invalidPersonData() throws Exception {
 
     }
 
+    //@Test
+    //public void execute_add_successful() throws Exception {
+    //    // setup expectations
+    //    TestDataHelper helper = new TestDataHelper();
+    //    Person toBeAdded = helper.adam();
+    //    AddressBook expectedAB = new AddressBook();
+    //    expectedAB.addPerson(toBeAdded);
+
+    //    // execute command and verify result
+    //    assertCommandBehavior(helper.generateAddCommand(toBeAdded),
+    //                          String.format(AddCommand.MESSAGE_SUCCESS, toBeAdded),
+    //                          expectedAB,
+    //                          false,
+    //                          Collections.emptyList());
+
+    //}
+
+
     @Test
-    public void execute_add_successful() throws Exception {
-        // setup expectations
+    public void execute_add_showsAllPersons() throws Exception {
+        // prepare expectations
         TestDataHelper helper = new TestDataHelper();
-        Person toBeAdded = helper.adam();
-        AddressBook expectedAB = new AddressBook();
-        expectedAB.addPerson(toBeAdded);
+        AddressBook expectedAB = helper.generateAddressBook(false, true);
+        List<? extends ReadOnlyPerson> expectedList = expectedAB.getAllPersons().immutableListView();
 
-        // execute command and verify result
-        assertCommandBehavior(helper.generateAddCommand(toBeAdded),
-                              String.format(AddCommand.MESSAGE_SUCCESS, toBeAdded),
-                              expectedAB,
-                              false,
-                              Collections.emptyList());
+        // prepare address book state
+        helper.addToAddressBook(addressBook, false, true);
+
+        assertCommandBehavior("list",
+                Command.getMessageForPersonListShownSummary(expectedList),
+                expectedAB,
+                true,
+                expectedList);
```

Comment:
Good job in testing this scenario. Try to see if other exception cases can be tested too.